mlflow/utils/requirements_utils.py

```diff
@@ -197,7 +197,7 @@ def _prune_packages(packages):
     """
     packages = set(packages)
     requires = set(_flatten(map(_get_requires_recursive, packages)))
-    return packages - requires
+    return packages - (requires - {"scikit-learn"})
```

Comment:
Is there a way we can make this a bit more generic? I.e. if the package is a dependency of MLflow, still include the package in the results?

Reply:
Yes. I think we can do that.